pkg/querier/builder_query.go

```diff
@@ -375,3 +410,18 @@ func decodeCursor(cur string) (int64, error) {
 	}
 	return strconv.ParseInt(string(b), 10, 64)
 }
+
+func seed(promoted map[string]any, body map[string]any) {
```

Comment:
depth limit should be added here else it will traverse everything.

Reply:
It needs to, else data will not appear from promoted